src/jit/assertionprop.cpp

```diff
@@ -2852,16 +2852,15 @@ GenTreePtr Compiler::optAssertionProp_LclVar(ASSERT_VALARG_TP assertions, const
     }
 
     BitVecOps::Iter iter(apTraits, assertions);
-    unsigned        index = 0;
-    while (iter.NextElem(&index))
+    for (int i = iter.NextElem(); i != -1; i = iter.NextElem())
     {
```

Comment:
I would prefer that we not use assignment as an expression. If we want to avoid the visual noise of the repeated calls to `NextElem()`, we should consider using `C++11 iterators.